Implement Attach debugger + picker

RemcoSmitsDev opened the pull request:
https://github.com/user-attachments/assets/29a80cd4-013a-4192-82d2-919ae872750f



## Todo

- [x] Rework how we and where we store the process id
  - Maybe store the process id on the config level
- [x] Shutdown debug client when reverse request `startDebugging` was handled
- [x] Clean up handle_debug_client_started
- [x] Show error message when adapter/client does not support attach

RemcoSmitsDev committed:
Rename StopDebugAdapters to ShutdownDebugAdapters

RemcoSmitsDev committed:
Remove debug config from install methods

RemcoSmitsDev committed:
Move requests methods to background executor

RemcoSmitsDev committed:
Wip attach with picker

RemcoSmitsDev committed:
Move find client to the first line of the method

The client should always be there at this point.

RemcoSmitsDev committed:
Fix correctly determine when to restart

While debugging an reverse request issue, the top level client did send terminated event with `restart: false` but we tried to restart the client resulting in the client never being cleaned up by us.

Because before this change we always assumed if we got a json value we are going to restart the client, which is wrong.

We no try to restart the client if:
- restart arg is a boolean and its true
- restart arg is a json value but no boolean

RemcoSmitsDev committed:
Clean up response to adapter

RemcoSmitsDev committed:
Merge branch 'main' into debugger

RemcoSmitsDev committed:
Merge branch 'debugger' into attach-debugger

RemcoSmitsDev committed:
Merge branch 'debugger' into attach-debugger

RemcoSmitsDev committed:
Fix clippy errors

Anthony-Eid commented:
> Show error message when adapter/client does not support attach

It would be cool if we could show the error message directly within `.zed/debug.json` when a user creates their debug config.

RemcoSmitsDev committed:
WIP tasks

Anthony-Eid committed:
Simplified debug task schema

This changes debug.json to look for adapter: adapter_name instead of
and object when a user selects a debug adatper and fixes the default
behavior of request (to launch)

Co-authored-by: Remco Smits <djsmits12@gmail.com>

RemcoSmitsDev committed:
Make default and flatten work for request

RemcoSmitsDev committed:
Rename enum case

RemcoSmitsDev committed:
Remove dbg

RemcoSmitsDev committed:
Dismiss when candidate is not found

RemcoSmitsDev committed:
Merge branch 'debugger' into attach-debugger

RemcoSmitsDev committed:
Add docs for why we update the process id on the config

RemcoSmitsDev committed:
Show error when `attach` request is selected but not supported
